testing/test_capture.py

```diff
@@ -642,6 +642,13 @@ def test_unicode_and_str_mixture(self):
             f.close()
             assert isinstance(s, unicode)
 
+    def test_write_bytes_to_buffer(self):
+        f = capture.CaptureIO()
+        # In python3, stdout / stderr are text io wrappers (exposing a buffer
+        # property of the underlying bytestream).
```

Comment:
I would rather at this point to be explicit about the expected behavior in Python 3:

```python
@pytest.mark.skipif(sys.version_info[0] == 2, 'python 3 only behavior')
def test_write_bytes_to_buffer(self):
    """In python3, stdout / stderr are text io wrappers (exposing a buffer
    property of the underlying bytestream). See issue #1407
    """
    f = capture.CaptureIO()
    f.buffer.write(b'foo\r\n')
    assert f.getvalue() == 'foo\n'
```

(I also added a `\r\n` there to ensure we are not messing with how newlines are treated)
    
```